### ISSUE TITLE
Fails to detect post name if name is included as img alt text

### ISSUE THREAD

**Issue by aaronpk** [aaronpk/XRay#57]
This might be a problem specific to Known's markup. For photo posts, Known includes a `p-name` tag with the caption entered for the photo. Separately (outside that tag), Known includes the photo's `img class=u-photo` tag inside `e-content`, and there is *also* an `alt` attribute on the image with the same text as in the `p-name`.

This causes XRay to first dedupe content/name, finding that they match, and dropping the `name` value. Then XRay removes the img tag from the content since it has `u-photo`, and then there is no text left in the content, and the name has been dropped, so the caption has been lost entirely.

Part of #52 again.

## html
```html
<html>
  <head>
    <title>Test</title>
  </head>
  <body class="h-entry">
    <h2 class="p-name">Photo caption</h2>
    <p class="e-content"><img class="u-photo" src="/photo.jpg" alt="Photo caption"></p>
  </body>
</html>
```

## mf2 json
```
{
    "type": [
        "h-entry"
    ],
    "properties": {
        "name": [
            "Photo caption"
        ],
        "photo": [
            "/photo.jpg"
        ],
        "content": [
            {
                "html": "<img class=\"u-photo\" src=\"/photo.jpg\" alt=\"Photo caption\">",
                "value": "Photo caption"
            }
        ]
    }
}
```

## xray result (in-progress)
```json
{
    "type": "entry",
    "photo": [
        "http://sanitize.example/photo.jpg"
    ],
    "content": {
        "text": null
    }
}
```

**Comment by aaronpk** [comment]
Fixed, new result is

```json
{
    "type": "entry",
    "photo": [
        "http://sanitize.example/photo.jpg"
    ],
    "name": "Photo caption"
}
```
